okhttp/src/main/kotlin/okhttp3/Headers.kt

```diff
@@ -271,6 +271,14 @@ class Headers private constructor(
       addLenient(name, value)
     }
 
+    /**
+     * Add a header with the specified name and value. Doesn't validate neither header names or values,
+     * allowing non-ASCII characters in both.
+     */
+    fun addUnsafeNonAsciiNameOrValue(name: String, value: String) = apply {
```

Comment:
Lets just change `addUnsafeNonAscii()` to not call `checkName()` and be done with it?

Reply:
Do you absolutely need trimming? I think that if you wanna go down the path of fuzzing at the HTTP protocol layer you might be better doing that with a raw TCP socket instead of a high-level client like OkHttp.

Reply:
That might be the ultimate solution, using raw TCP socket connections. Currently though, with the existing fuzzers, OkHttp is good enough (with the changes from this PR). I'm currently using a fork, but it will be great to use the official version.
There are fuzzers which prefix/suffix header values with spaces, thus having a method that doesn't do trimming is important.

If `checkName()` is removed maybe we can have 2 methods `addUnsafeNonAscii()` that does the trimming and add another one `addUnsafeNonAsciiNoTrim()`?

Reply:
I’d really like to keep trimming mandatory in OkHttp. I don’t like the consequences of permitting even more weird `Headers` instances.

Reply:
Ok, I can appreciate keeping things stricter. In this case, would it be ok then just to remove `checkName()`?

Reply:
Yep! That’s a good compromise.

Reply:
New method removed and also removed call to `checkName()`